securedrop_client/gui/widgets.py

```diff
@@ -975,11 +994,13 @@ def update(self, sources: List[Source]) -> List[str]:
                 new_source = SourceWidget(self.controller, source)
                 self.source_widgets[source.uuid] = new_source
 
-                list_item = QListWidgetItem()
+                list_item = SourceListWidgetItem()
                 self.insertItem(0, list_item)
                 list_item.setSizeHint(new_source.sizeHint())
                 self.setItemWidget(list_item, new_source)
 
+        # Sort..!
+        self.sortItems(Qt.DescendingOrder)
```

Comment:
since we're resorting the entire source list for every sync, we'll need to test this against 400 and 1000 sources to make sure things don't freeze up